my_work_R/my_happy_script.R

```diff
@@ -1,93 +1,101 @@
-  mutate(dl = -pgdl, pb = 0, n_prof = n_profs)
-
-
-for (mod in c('pb','dl','pgdl')){
-  mod_data <- filter(eval_data, model_type == mod)
-  mod_profiles <- unique(mod_data$n_prof)
-  for (mod_profile in mod_profiles){
-    d <- filter(mod_data, n_prof == mod_profile) %>% summarize(y0 = min(rmse), y1 = max(rmse), col = unique(col))
-    x_pos <- offsets %>% filter(n_prof == mod_profile) %>% pull(!!mod) + mod_profile
-    lines(c(x_pos, x_pos), c(d$y0, d$y1), col = d$col, lwd = 2.5)
-  }
-  d <- group_by(mod_data, n_prof) %>% summarize(y = mean(rmse), col = unique(col), pch = unique(pch)) %>%
-    rename(x = n_prof) %>% arrange(x)
-  
-  lines(d$x + tail(offsets[[mod]], nrow(d)), d$y, col = d$col[1], lty = 'dashed')
-  points(d$x + tail(offsets[[mod]], nrow(d)), d$y, pch = d$pch[1], col = d$col[1], bg = 'white', lwd = 2.5, cex = 1.5)
-  
-}
-
-points(2.2, 0.79, col = '#7570b3', pch = 23, bg = 'white', lwd = 2.5, cex = 1.5)
-text(2.3, 0.80, 'Process-Guided Deep Learning', pos = 4, cex = 1.1)
-
-points(2.2, 0.94, col = '#d95f02', pch = 22, bg = 'white', lwd = 2.5, cex = 1.5)
-text(2.3, 0.95, 'Deep Learning', pos = 4, cex = 1.1)
-
-points(2.2, 1.09, col = '#1b9e77', pch = 21, bg = 'white', lwd = 2.5, cex = 1.5)
-text(2.3, 1.1, 'Process-Based', pos = 4, cex = 1.1)
-
-dev.off()
+eval_data <- readr::read_csv(mendota_file, col_types = 'iccd') |>
+  filter(str_detect(exper_id, 'similar_[0-9]+')) |>
+  group_by(exper_id, model_type) |> 
+  summarize(
+    mean_rmse = mean(rmse),
+    min_rmse = min(rmse),
+    max_rmse = max(rmse),
+    ct_exper = n()
+  )
+
+eval_data <- eval_data |> 
+  mutate(
+    col = case_when(
+      model_type == 'pb' ~ '#1b9e77',
+      model_type == 'dl' ~'#d95f02',
+      model_type == 'pgdl' ~ '#7570b3'
+      ), 
+    pch = case_when(
+      model_type == 'pb' ~ 21,
+      model_type == 'dl' ~ 22,
+      model_type == 'pgdl' ~ 23
+      ), 
+    n_prof = as.numeric(str_extract(exper_id, '[0-9]+')),
+    model_type = factor(model_type, levels = c("pgdl", "dl", "pb"))
+    ) |> 
+    mutate(
+      model_legend = fct_recode(
+        model_type,
+        "Process-Guided Deep Learning" = "pgdl",
+        "Deep Learning" = "dl",
+        "Process-Based" = "pb"
+      )
+    )
+
+g <- 
+  ggplot(data = eval_data, 
+         aes(color = model_legend, shape = model_legend)) +
+  geom_line(aes(x = n_prof, y = mean_rmse), 
+            linetype = "dashed",
+            lwd = 0.5,
+            position = position_dodge(width = 0.05)) +
+  geom_linerange(aes(x = n_prof, ymin = min_rmse, ymax = max_rmse),
+                 lwd = 0.5,
+                 position = position_dodge(width = 0.05)) +
+  geom_point(aes(x = c(2, 10, 50, 100, 500, 980), y = mean_rmse), 
+             position = position_dodge(width = 0.05), 
+             fill = "white",
+             size = 3) +
+  labs(
+    x = "Training Temperature Profiles (#)",
+    y = "Test RMSE (°C)"
+  )
+
+g <- g +
+  scale_x_log10(breaks = n_profs, limits = c(1, 1000)) +
```

Comment:
Also suggest dropping the `limits` here to make the spacing between the y-axis and the chart elements a little closer